libs/ui-components/src/components/Device/DevicesPage/DecommissionedDevicesTable.tsx

```diff
@@ -0,0 +1,156 @@
+import TablePagination from '../../Table/TablePagination';
+import MassDeleteDeviceModal from '../../modals/massModals/MassDeleteDeviceModal/MassDeleteDeviceModal';
+import AddDeviceModal from '../AddDeviceModal/AddDeviceModal';
+import { DecommissionedDevicesEmptyState } from './DevicesEmptyStates';
+import DecommissionedDeviceTableRow from './DecommissionedDeviceTableRow';
+import DeviceNameOnlyToolbarFilter from './DeviceNameOnlyToolbarFilter';
+
+interface DecommissionedDevicesTableProps {
+  devices: Array<Device>;
+  refetch: VoidFunction;
+  nameOrAlias: string | undefined;
+  setOnlyDecommissioned: (check: boolean) => void;
+  setNameOrAlias: (text: string) => void;
+  hasFiltersEnabled: boolean;
+  isFilterUpdating: boolean;
+  pagination: Pick<PaginationDetails<DeviceList>, 'currentPage' | 'setCurrentPage' | 'itemCount'>;
+}
+
+const getDeviceColumns = (t: TFunction): ApiSortTableColumn[] => [
+  {
+    name: t('Name'),
+  },
+  {
+    name: t('Status'),
+  },
+  {
+    name: t('Last seen'),
+  },
+];
+
+const DecommissionedDevicesTable = ({
+  devices,
+  refetch,
+  nameOrAlias,
+  setNameOrAlias,
+  hasFiltersEnabled,
+  setOnlyDecommissioned,
+  isFilterUpdating,
+  pagination,
+}: DecommissionedDevicesTableProps) => {
+  const { t } = useTranslation();
+  const [addDeviceModal, setAddDeviceModal] = React.useState(false);
+  const [isMassDeleteModalOpen, setIsMassDeleteModalOpen] = React.useState(false);
+  const { remove } = useFetch();
+  const deviceColumns = React.useMemo(() => getDeviceColumns(t), [t]);
+
+  const { onRowSelect, hasSelectedRows, isAllSelected, isRowSelected, setAllSelected } = useTableSelect();
+
+  const { action: deleteDeviceAction, modal: deleteDeviceModal } = useDeleteListAction({
+    resourceType: 'Device',
+    onConfirm: async (resourceId: string) => {
+      await remove(`devices/${resourceId}`);
+      refetch();
+    },
+  });
+
+  const [canDelete] = useAccessReview(RESOURCE.DEVICE, VERB.DELETE);
+  const [canEdit] = useAccessReview(RESOURCE.DEVICE, VERB.PATCH);
+
+  return (
+    <>
+      <Toolbar id="decommissioned-devices-toolbar" inset={{ default: 'insetNone' }}>
+        <ToolbarContent>
+          <ToolbarGroup>
+            <ToolbarItem variant="search-filter">
+              <DeviceNameOnlyToolbarFilter nameOrAlias={nameOrAlias} setNameOrAlias={setNameOrAlias} />
+            </ToolbarItem>
+            <ToolbarItem>
+              <Button aria-label={t('Add devices')} onClick={() => setAddDeviceModal(true)}>
+                {t('Add devices')}
+              </Button>
+            </ToolbarItem>
+            {canDelete && (
+              <ToolbarItem>
+                <Button
+                  isDisabled={!hasSelectedRows}
+                  onClick={() => setIsMassDeleteModalOpen(true)}
+                  variant="secondary"
+                >
+                  {t('Delete forever')}
+                </Button>
+              </ToolbarItem>
+            )}
+            <ToolbarItem>
```

Comment:
```suggestion
            <ToolbarItem alignSelf="center">
```

applies also for the second instance in `EnrolledDevicesTable.ts`